appservice/src/deploy/runDeployTask.ts

```diff
@@ -70,9 +71,9 @@ export interface IPreDeployTaskResult {
     failedToFindTask: boolean;
 }
 
-function shouldExecuteTask(scmType: string | undefined, settingKey: string, taskName: string): boolean {
+function shouldExecuteTask(context: IDeployContext, scmType: string | undefined, settingKey: string, taskName: string): boolean {
     // We don't run deploy tasks for non-zipdeploy since that stuff should be handled by kudu
-    const shouldExecute: boolean = scmType !== ScmType.LocalGit && scmType !== ScmType.GitHub;
+    const shouldExecute: boolean = context.deployMethod === 'storage' || context.deployMethod === 'zip' || (scmType !== ScmType.LocalGit && scmType !== ScmType.GitHub);
```

Comment:
Did this boolean expression need to change?  Wouldn't `scmType !== ScmType.LocalGit && scmType !== ScmType.GitHub` still cover this (since if the deployMethod were `storage` or `zip` then scmType would be `ScmType.None`)?

Reply:
deployMethod is "Used to overwrite default deploy method based on scm type"

So like Linux Consumption doesn't support LocalGit/GitHub, so we always use zip regardless of scmType